### PR TITLE
Move deployer to a new module

### PR DESCRIPTION
**Pull request by thusithathilina**
This moves the MSF4J Deployer to a separate module and wraps that with a new feature. Resolves https://github.com/wso2/msf4j/issues/428

### CODE REVIEW

File: deployer/src/main/java/org/wso2/msf4j/deployer/MicroservicesDeployer.java [hunk 1]
```diff
@@ -48,12 +39,7 @@
  * This will be picked by the DeploymentEngine service according to the whiteboard pattern to deploy
  * micro service POJO artifacts.
  */
-@Component(
-        name = "org.wso2.carbon.mss.internal.deployer.MicroservicesDeployer",
-        service = Deployer.class,
-        immediate = true
-)
-public class MicroservicesDeployer implements Deployer, RequiredCapabilityListener {
```
Comment (afkham) [inline deployer/src/main/java/org/wso2/msf4j/deployer/MicroservicesDeployer.java]: Why was RequiredCapabilityListener removed?

File: deployer/src/main/java/org/wso2/msf4j/deployer/MicroservicesDeployer.java [hunk 2]
```diff
@@ -13,26 +13,17 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.wso2.msf4j.internal.deployer;
```
Comment (afkham) [inline deployer/src/main/java/org/wso2/msf4j/deployer/MicroservicesDeployer.java]: Why was this exported from the package? (Not private)

File: deployer/src/main/java/org/wso2/msf4j/deployer/internal/DataHolder.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (imesh) [inline deployer/src/main/java/org/wso2/msf4j/deployer/internal/DataHolder.java]: Might be better to change the year to 2017.

File: deployer/src/test/java/org/wso2/msf4j/deployer/MSF4JDeployerTest.java [hunk 4]
```diff
@@ -56,6 +55,8 @@
  */
 public class MSF4JDeployerTest {
 
+    public static final String HOSTNAME = "localhost";
+    public static final int PORT = 8090;
```
Comment (imesh) [inline deployer/src/test/java/org/wso2/msf4j/deployer/MSF4JDeployerTest.java]: May be these variables can be made private?

**Pull request merged**